src/post/post.repository.knex.ts

```diff
@@ -19,6 +20,39 @@ export class KnexPostRepository {
         private readonly events: EventEmitter,
     ) {}
 
+    private mapRowToPost(row: any): Post {
+        const author = new Account(
+            row.author_id,
+            row.username,
+            row.name,
+            row.bio,
+            parseURL(row.avatar_url),
+            parseURL(row.banner_image_url),
+            null,
+        );
+
+        return new Post(
+            row.id,
+            row.uuid,
+            author,
+            row.type,
+            row.audience,
+            row.title,
+            row.excerpt,
+            row.content,
+            new URL(row.url),
+            parseURL(row.image_url),
+            new Date(row.published_at),
+            row.like_count,
+            row.repost_count,
+            row.reply_count,
+            row.in_reply_to,
```

Comment:
This is required to be of type `Post | null` - and same for the `threadRoot` param, but here it will be a number.

I'm not sure if we want to eagerly load the parent posts, because we could end up with a lot of data (and queries) when fetching posts which are deep in a chain of replies 🤔 

I think it would be better to have some methods either on the repository or the service which can fetch the parent, root or entire chain for posts 🤔 

This is a fairly big change so I'm gonna take a look at them as you're off today

Reply:
Thanks, I think it will be good to have those methods. I needed that to set the threadRoot [here](https://github.com/TryGhost/ActivityPub/pull/321/files#diff-c77b89ae7ba2627f28f4b8fe20ecbb5c169809193c3a3cfd404cd93f12750b83R307) but I ended up making two queries for it.